src/errorHandling.ts

```diff
@@ -192,6 +192,52 @@ export function getErrorHandledSync<T extends (...args: any[]) => any>(
   };
 }
 
+function getFeatureFlags(): string[] {
+  const getBooleanFields = (docName: string) => {
+    const doc = fyo.singles[docName];
+    if (!doc) return {};
+
+    return Object.entries(doc).reduce((acc, [key, value]) => {
+      const field = fyo.schemaMap[docName]?.fields.find(
+        (f) => f.fieldname === key
+      );
```

Comment:
Here, `find()`  runs once per key. If the number of fields is large, this creates O(n^2) complexity. 

You can fix this by building a Map of (fieldname, field) before the loop 

eg: 
```ts
const fieldsArray = fyo.schemaMap[docName]?.fields ?? [];
const fieldsMap = new Map(fieldsArray.map(f => [f.fieldname, f]));

```

and you can access the items from the map easily like, 
```ts
const field = fieldsMap.get(key);
```

[Map Reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)